process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+    
+    """
+    GSTT_urls = ""
+    for index, row in df.iterrows():
+        if not bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            GSTT_urls += line
+    return GSTT_urls
+
+def get_data_StG(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links for files destined for StG transfer and creates a string argument.
+    
+    """
+    StG_urls = ""
+    for index, row in df.iterrows():
+        if bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            StG_urls += line
+    return StG_urls
+
+def save_log_file(text, filename): 
+    cur_time = datetime.now()
+    cur_time_string = cur_time.strftime("%Y_%m_%d__%H_%M_%S_")
+    log_path = "{}\\process_logs\\Finished_on_{}_{}.txt".format(config.CSVread_folder, cur_time_string, filename)
+    with open(log_path, 'w') as f:
+        f.write(text)
+
+def get_files():
+    """
+    This function gets the paths for csv files download by duty bioinformatician to the Duty_Bioinformatics_CSV folder.
+    """
+    folder = config.CSVread_folder
+    files = (
+        file
+        for file in os.listdir(folder)
+        if os.path.isfile(os.path.join(folder, file))
+    )
+    files_list = collections.defaultdict(list)
+    for filename in files:
+        results = re.search("__(\S+)__", filename)
+        project = results.group(1)
+        filepath = folder + "\\" + filename
+        files_list[project].append(filepath)
+    files_merged = dict(files_list)
+    return files_merged
+
+class Project:
+    """
+    Process project
+    """
+    def __init__(self, project, files):
+        self.project = project
+        self.project_files = files
+    def process_data(self):
+        """
+        This function executes the processing of a project based on the project type e.g. SNP, WES, TSO500 or MokaPipe
+        """
+        if self.project == "WES":
+            Project.process_WES(self)
+        elif self.project == "SNP":
+            Project.process_SNP(self)
+        elif self.project == "TSO500":
+            Project.process_TSO(self)
+        elif self.project == "MokaPipe":
+            Project.process_MokaPipe(self)
+    def process_WES(self):
+        """
+        Process WES Runs
+        """
+        destination_of_files = config.WES_destination_of_files
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            url_wes = get_data(csv_file, '"{}"'.format(destination_of_files))
+            output = download_data(url_wes)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_SNP(self):
+        """
+        Process SNP Runs
+        """
+        destination_of_VCFs = config.SNP_destination_of_VCFs

```

Comment:
No need to make a new variable here - could just use config.SNP... below

Reply:
changed that